### PR TITLE
Cppcheck/v2

### PR DESCRIPTION
**Pull request by victorjulien**
Cppcheck 2.14 fixes plus a few misc improvements.

Replaces #11946, implementing review comments.

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 23109

File: src/tm-threads.c [hunk 1]
```diff
@@ -2190,14 +2190,26 @@ void TmThreadsSetThreadTimestamp(const int id, const SCTime_t ts)
     int idx = id - 1;
     Thread *t = &thread_store.threads[idx];
     t->pktts = ts;
-    struct timeval systs;
-    gettimeofday(&systs, NULL);
-    t->sys_sec_stamp = (uint32_t)systs.tv_sec;
+    struct timeval tv;
```
Comment (jufajardini) [inline src/tm-threads.c:2193]: I see that in same situations we're still using `timeval`. What's the criteria, now? Is `timeval` only for local variables, and for setting or initializing an `SCTime_t`?

File: src/tm-threads.c [hunk 2]
```diff
@@ -2190,14 +2190,26 @@ void TmThreadsSetThreadTimestamp(const int id, const SCTime_t ts)
...
+    }
+#endif
+
+    t->sys_sec_stamp = SCTIME_FROM_TIMEVAL(&tv);
     SCMutexUnlock(&thread_store_lock);
 }
 
 bool TmThreadsTimeSubsysIsReady(void)
 {
+    static SCTime_t nullts;
```
Comment (jufajardini) [inline src/tm-threads.c:2212]: `SCTIME_INITIALIZER` not needed in this case?

**Review by jufajardini — COMMENTED** [review]
Seems way more `time`-related than `cpp` :P

Noticed that in one case we're not using `SCTIME_INITIALIZER`, not sure if relevant.

File: src/tm-threads.c (hunk 1 above)
Comment (victorjulien) [reply src/tm-threads.c:2193]: we still need it for gettimeofday

File: src/tm-threads.c (hunk 2 above)
Comment (victorjulien) [reply src/tm-threads.c:2212]: I'll add it

**Comment by victorjulien** [comment]
replaced by #12027

**Pull request closed without merging**